phalcon/mvc/model/query/builder.zep

```diff
@@ -861,15 +861,325 @@ class Builder implements BuilderInterface, InjectionAwareInterface
+	 *
+	 *<code>
+	 *  $builder->andHaving('name = "Peter"');
+	 *  $builder->andHaving('name = :name: AND id > :id:', array('name' => 'Peter', 'id' => 100));
+	 *</code>
+	 *
+	 * @param string conditions
+	 * @param array bindParams
+	 * @param array bindTypes
+	 * @return \Phalcon\Mvc\Model\Query\Builder
+	 */
+	public function andHaving(string! conditions, var bindParams = null, var bindTypes = null) -> <Builder>
+	{
+		var currentBindParams, currentBindTypes, mergedParams,
+			mergedTypes, currentConditions, newConditions;
+
+		let currentConditions = this->_having;
+
+		/**
+		 * Nest the condition to current ones or set as unique
+		 */
+		if currentConditions {
+			let newConditions = "(" . currentConditions . ") AND (" . conditions . ")";
+		} else {
+			let newConditions = conditions;
+		}
+		let this->_having = newConditions;
+
+		/**
+		 * Merge the bind params to the current ones
+		 */
+		if typeof bindParams == "array" {
+			let currentBindParams = this->_bindParams;
+			if typeof currentBindParams == "array" {
+				let mergedParams = currentBindParams + bindParams;
+			} else {
+				let mergedParams = bindParams;
+			}
+			let this->_bindParams = mergedParams;
+		}
+
+		/**
+		 * Merge the bind types to the current ones
+		 */
+		if typeof bindTypes == "array" {
+			let currentBindTypes = this->_bindTypes;
+			if typeof currentBindParams == "array" {
+				let mergedTypes = currentBindTypes + bindTypes;
+			} else {
+				let mergedTypes = bindTypes;
+			}
+			let this->_bindTypes = mergedTypes;
+		}
+
+		return this;
+	}
+
+	/**
+	 * Appends a condition to the current having conditions using a OR operator
+	 *
+	 *<code>
+	 *  $builder->orHaving('name = "Peter"');
+	 *  $builder->orHaving('name = :name: AND id > :id:', array('name' => 'Peter', 'id' => 100));
+	 *</code>
+	 *
+	 * @param string conditions
+	 * @param array bindParams
+	 * @param array bindTypes
+	 * @return \Phalcon\Mvc\Model\Query\Builder
+	 */
+	public function orHaving(string! conditions, var bindParams = null, var bindTypes = null) -> <Builder>
+	{
+		var currentBindParams, currentBindTypes, mergedParams,
+			mergedTypes, currentConditions;
+
+		/**
+		 * Nest the condition to current ones or set as unique
+		 */
+		let currentConditions = this->_having;
+		if currentConditions {
+			let this->_having = "(" . currentConditions . ") OR (" . conditions . ")";
+		} else {
+			let this->_having = conditions;
+		}
+
+		/**
+		 * Merge the bind params to the current ones
+		 */
+		if typeof bindParams == "array" {
+			let currentBindParams = this->_bindParams;
+			if typeof currentBindParams == "array" {
+				let mergedParams = currentBindParams + bindParams;
+			} else {
+				let mergedParams = bindParams;
+			}
+			let this->_bindParams = mergedParams;
+		}
+
+		/**
+		 * Merge the bind types to the current ones
+		 */
+		if typeof bindTypes == "array" {
+			let currentBindTypes = this->_bindTypes;
+			if typeof currentBindParams == "array" {
```

Comment:
I think so. I just duplicated the where functions. Having function should behave like where so I guess everything is fine.